### DISCUSSION TITLE
ServiceDiscovery: Explain Scheme selection when resolving HTTP(S) endpoints

### DISCUSSION THREAD

**Discussion by cloudfy** [microsoft/aspire#6031] [Q&A]
Can anyone explain the service discovery when dealing with multi scheme addressing; for example: "https+http://basket". 

When implementing a custom IServiceEndpointProvider, I somehow sometimes get the non-HTTPS and the HTTPS endpoints mixed up in the service selection if the (example) basket API is operating on two schemes, non-HTTPS and the HTTPS.

Example:
- Basket API > Registers in the service registry with two endpoints (one http and one https). Different ports.
- Custom IServiceEndpointProvider resolves two endpoints and return into the collection.
- Using the custom *ServiceEndpointProvider, discovery of the HTTPS endpoint resolve a DnsEndpoint of the non-HTTP endpoint causing the API not to be contactable.

Aspire has the UriEndPoint class which hold the scheme (and can be used to match), however this is internal and any custom implementation is not usable and this is not implemented into the ServiceDiscovery abstract nuget.

**Comment by davidfowl** [comment]
cc @ReubenBond

**Reply by cloudfy** [reply]
Anyone?

**Reply by cloudfy** [reply]
I could really use some experts here... @davidfowl  anyone else on the project with knowledge?
